### ISSUE TITLE
[Bug]: Addon-a11y postinstall does not update vitest setup

### ISSUE THREAD

**Issue by shilman** [storybookjs/storybook#32242]
### Describe the bug

If you have the Vitest addon install and you run:

```sh
storybook add @storybook/addon-a11y
```

It will add the addon to your `main.js` but won't add it to your `vitest.setup.ts` file, which means that a11y tests won't run in Vitest CLI or via the UI. What's worse is that this all fails silently.

Ideally we should add `@storybook/addon-a11y` to `vitest.setup.ts`. However, we don't know the structure of the file so that could be hard. If it's not feasible, we should detect the existence of `vitest.setup.ts` and tell the user how to add by hand:

```diff
+ import * as a11yAnnotations from "@storybook/addon-a11y/preview";

- setProjectAnnotations([projectAnnotations]);
+ setProjectAnnotations([projectAnnotations, a11yAnnotations]);
```

NOTE: This will no longer be necessary once [CSF Factories](https://github.com/storybookjs/storybook/discussions/30112) is widespread. However, that will take time so in the meantime we should do this instead.

### Reproduction link

N/A

### System

```bash
N/A
```

**Comment by shilman** [comment]
Can we also do this as part of the CLI project? @ghengeveld @valentinpalkovic

**Comment by ghengeveld** [comment]
@shilman Adding it to stretch goals.
